block/produce.go

```diff
@@ -66,12 +69,28 @@ func (m *Manager) ProduceBlockLoop(ctx context.Context) {
 				continue
 			}
 			resetEmptyBlocksTimer()
-		case shouldProduceBlocks := <-m.shouldProduceBlocksCh:
-			for !shouldProduceBlocks {
-				m.logger.Info("block production paused - awaiting positive continuation signal")
-				shouldProduceBlocks = <-m.shouldProduceBlocksCh
+
+			// Check if we should submit the accumulated data
+			if m.shouldSubmitBatch() {
+				select {
+				case m.ShouldSubmitBatchCh <- true:
+					m.logger.Info("new batch accumulated, signal sent to submit the batch")
+				default:
+					m.logger.Error("new batch accumulated, but channel is full, stopping block production until the signal is consumed")
+					// emit unhealthy event for the node
+					evt := &events.DataHealthStatus{Error: fmt.Errorf("submission channel is full")}
+					uevent.MustPublish(ctx, m.Pubsub, evt, events.HealthStatusList)
+					// wait for the signal to be consumed
+					m.ShouldSubmitBatchCh <- true
+					m.logger.Info("resumed block production")
+					// emit healthy event for the node
+					evt = &events.DataHealthStatus{Error: nil}
+					uevent.MustPublish(ctx, m.Pubsub, evt, events.HealthStatusList)
+				}
+				m.produceBlockMutex.Lock()
+				m.AccumulatedProducedSize.Store(0)
+				m.produceBlockMutex.Unlock()
```

Comment:
it refactored in the next PR

Reply:
hmm ok
it might be better not to merge to main though, with known bugs like what Sergi raised https://github.com/dymensionxyz/dymint/pull/794#discussion_r1592330995 and what I put above